core/src/main/java/com/linecorp/armeria/server/DefaultServerErrorHandler.java

```diff
@@ -86,8 +89,24 @@ public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable caus
         }
 
         if (cause instanceof RequestTimeoutException) {
-            return internalRenderStatus(ctx, ctx.request().headers(),
-                                        HttpStatus.SERVICE_UNAVAILABLE, cause);
+            final HttpStatus status;
+            if (ctx.method() == HttpMethod.GET) {
+                // Consider GET requests as having received the request fully.
+                status = HttpStatus.SERVICE_UNAVAILABLE;
+            } else {
+                final RequestContextExtension ctxExtension = ctx.as(RequestContextExtension.class);
+                assert ctxExtension != null;
+                final Request request = ctxExtension.originalRequest();
```

Comment:
Optional) Because I was second guessing whether there is a case `ctx.originalRequest` isn't a `DecodedHttpRequest`

```suggestion
                final DecodedHttpRequest request = (DecodedHttpRequest) ctxExtension.originalRequest();
```

Reply:
Thanks fixed. 😉